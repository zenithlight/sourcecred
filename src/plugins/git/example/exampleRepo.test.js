// @flow

import tmp from "tmp";

import {
  createExampleRepo,
  createExampleSubmoduleRepo,
  SUBMODULE_COMMIT_1,
  SUBMODULE_COMMIT_2,
} from "./exampleRepo";

const cleanups: (() => void)[] = [];
afterAll(() => {
  cleanups.forEach((f) => {
    f();
  });
});

function mkdtemp() {
  const result = tmp.dirSync({unsafeCleanup: true});
  cleanups.push(() => result.removeCallback());
  return result.name;
}

describe("plugins/git/example/exampleRepo", () => {
  describe("createExampleRepo", () => {
    it("is deterministic", () => {
      expect(createExampleRepo(mkdtemp()).commits).toMatchSnapshot();
    });
  });

  describe("createExampleSubmoduleRepo", () => {
    it("is deterministic", () => {
      expect(createExampleSubmoduleRepo(mkdtemp()).commits).toMatchSnapshot();
    });

    it("includes all the SHAs that it should", () => {
      const commits = createExampleSubmoduleRepo(mkdtemp()).commits;
      expect(commits).toEqual(
        expect.arrayContaining([SUBMODULE_COMMIT_1, SUBMODULE_COMMIT_2])
      );
    });
  });
});
