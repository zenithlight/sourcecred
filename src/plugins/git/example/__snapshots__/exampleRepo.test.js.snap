// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugins/git/example/exampleRepo createExampleRepo is deterministic 1`] = `
Array [
  "c2b51945e7457546912a8ce158ed9d294558d294",
  "c08ee3a4edea384d5291ffcbf06724a13ed72325",
  "17eacd3f5e8e1e21eb88f3f37a74cdd8db19f9e4",
  "674f9211d5058037a0a4f899648b3a53ac267733",
  "c68e9837f02a303ae98f376457b52e8ec17daefc",
  "f95257474a66c097dcd580374048e0b7423634f7",
  "ab5f98da9bda6f1168e76d5057dc09dec08aa7f3",
  "be289e85543ee1feb91e30f2416d09bb1914db21",
]
`;

exports[`plugins/git/example/exampleRepo createExampleSubmoduleRepo is deterministic 1`] = `
Array [
  "762c062fbdc7ec198cd693e95d55b374a08ff3e3",
  "29ef158bc982733e2ba429fcf73e2f7562244188",
]
`;
